docs/fundamentals/code-analysis/quality-rules/ca2254.md

```diff
@@ -25,11 +25,40 @@ A message template passed to a logger API is not constant.
 
 ## Rule description
 
+When performing logging, we want to preserve the structure of the log (including parameter names) along with the parameter values themselves. This allows for better observability and search in log aggregation and monitoring software. We want to maintain the full fidelity of the log message.
+
+Therefore, we prefer:
+
+```csharp
+var firstName = "Bob";
+var lastName = "Smith";
```

Comment:
```suggestion
var lastName = "Otto";
```

Reply:
@gewarren very cool! Is that public? Love the idea.

Reply:
Unfortunately, no.